gensim/similarities/index.py

```diff
@@ -15,16 +19,32 @@
 
 class AnnoyIndexer(object):
 
-    def __init__(self, model, num_trees):
+    def __init__(self, model=None, num_trees=None):
+        self.index = None
+        self.labels = None
         self.model = model
         self.num_trees = num_trees
 
-        if isinstance(self.model, Doc2Vec):
-            self.build_from_doc2vec()
-        elif isinstance(self.model, Word2Vec):
-            self.build_from_word2vec()
-        else:
-            raise ValueError("Only a Word2Vec or Doc2Vec instance can be used")
+        if model and num_trees:
+            if isinstance(self.model, Doc2Vec):
+                self.build_from_doc2vec()
+            elif isinstance(self.model, Word2Vec):
+                self.build_from_word2vec()
+            else:
+                raise ValueError("Only a Word2Vec or Doc2Vec instance can be used")
+
+    def save(self, fname):
+        self.index.save(fname)
+        d = {'f': self.model.vector_size, 'num_trees': self.num_trees, 'labels': self.labels}
+        pickle.dump(d, open(fname+'.d', 'wb'), 2)
+
+    def load(self, fname):
+        if os.path.exists(fname) and os.path.exists(fname+'.d'):
+            d = pickle.load(open(fname+'.d', 'rb'))
```

Comment:
Please use smart_open as in https://github.com/RaRe-Technologies/gensim/blob/6a289fefd72f038c8cc14826f63624950f5de1f8/gensim/utils.py#L907